content/cpp/concepts/polymorphism/polymorphism.md

```diff
@@ -0,0 +1,115 @@
+---
+Title: 'Polymorphism'
+Description: 'The word Polymorphism means having many form. Polymorphism is an important concept of object-oriented programming. It simply means more than one form. That is, the same entity (function or operator) can behave differently in different scenarios.'
+Subjects:
+  - 'Computer Science'
+  - 'Game Development'
+Tags:
+  - 'Function'
+  - 'Polymorphism'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+Polymorphism is an important concept of object-oriented programming. It simply means more than one form. That is, the same entity (function or operator) can operate differently under different situations.
+
+In C++ we have two types of polymorphism:
+
+- Compile time Polymorphism – This is also called as static (or early) binding.
+- Runtime Polymorphism – This is also called as dynamic (or late) binding.
+
+## Compile time Polymorphism
+
+Compile time polymorphism is perfect for **function overloading** as well as **operator overloading**.
+
+### Compile time Polymorphism Example
+
+In below example, we have two functions with same name but different number of arguments. Based on the number of parameters we pass during function invocation (function calling) determines which function is to be called, this is why it is considered as an example of **polymorphism** because in different conditions the output is different. Since, the call is determined during **compile time** thats why it is called **compile time polymorphism**.
```

Comment:
```suggestion
In the below example, there are two functions with the same name but a different number of arguments. 

Based on the number of parameters we pass during function invocation (function calling) determines which function is to be called, this is why it is considered as an example of polymorphism because in different conditions the output is different. Since, the call is determined during compile time that's why it is called compile-time polymorphism.
```

Reply:
made the requested changes successfully.